### PR TITLE
add link to open response tab

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** 
When esg flag enable we:
- add additional column that is direct to esg app
- the staff would behave the same if esg flag enabled
- only display link based on the old logic

JIRA: 
- [AU-307](https://openedx.atlassian.net/browse/AU-307)
- [AU-452](https://openedx.atlassian.net/browse/AU-452)

Related PR: 
- [edx-platform](https://github.com/edx/edx-platform/pull/29029)

**What changed?**
- Sending `ORA_GRADING_MICROFRONTEND_URL` to the client side
- Open response would render link to `frontend-app-ora-grading` on flag enabled and just normal text on flag disabled

When ESG flag enable:
Screen Shot 2021-12-14 at 10 42 19 AM

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- [ ] pull and test [this pr](https://github.com/edx/edx-platform/pull/29029)
- [ ] enable `openresponseassessment.enhanced_staff_grader` for the course
- [ ] Go to instructor board -> open response tab. The url should show in the `Assessment` column that would lead you to ESG mfe.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/static/js/spec/lms/oa_listing.js [hunk 1]
```diff
@@ -296,4 +295,39 @@ describe("OpenAssessment.CourseItemsListingView", function() {
             dataType: "json"
         });
     });
+
+    describe('esg enchanced grade flag', function() {
+        it('is enablend', () => {
```
Comment (muselesscreator) [inline openassessment/xblock/static/js/spec/lms/oa_listing.js:300]:
```suggestion
        it('is enabled', () => {
```

**Review by muselesscreator — CHANGES_REQUESTED** [review]
minor change requests for typos and variable names

**Review by muselesscreator — APPROVED** [review]
LGTM

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 2]
```diff
@@ -52,12 +52,39 @@ export class CourseItemsListingView {
       url: 'url_grade_available_responses',
     });
 
+    const context = this.data.CONTEXT || {};
+
+    const esgEnabled = context.ENHANCED_STAFF_GRADER;
+    const esgUrl = context.ORA_GRADING_MICROFRONTEND_URL;
+
+    const UnitNameCell = Backgrid.UriCell.extend({
```
Comment (jansenk) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:60]: I would look at making this sort of change around line 30, where we set up the "click number to go to page" sort of thing. Of course since this is an external page we'll want to just make the link an actual link rather than what they're currently doing there.

Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:60]: The variable name `AssessmentCell` was what drove me into creating a new one. Additionally, it had a certain level on string comparing.

**Review by mattcarter — COMMENTED** [review]
link to ESG with the `staff` column

File: openassessment/xblock/static/js/spec/lms/oa_listing.js [hunk 3]
```diff
@@ -296,4 +295,37 @@ describe("OpenAssessment.CourseItemsListingView", function() {
             dataType: "json"
         });
     });
+
+
+    describe('esg enchanced grade flag', function() {
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/spec/lms/oa_listing.js:300]: Are we able to test here that the new column is present/absent?

File: openassessment/staffgrader/tests/test_decorators.py [hunk 4]
```diff
@@ -6,8 +6,8 @@
 from uuid import uuid4
 
 from mock import Mock
-from submissions.errors import SubmissionInternalError, SubmissionNotFoundError, SubmissionRequestError
 from xblock.exceptions import JsonHandlerError
+from submissions.errors import SubmissionInternalError, SubmissionNotFoundError, SubmissionRequestError
```
Comment (nsprenkle) [inline openassessment/staffgrader/tests/test_decorators.py:10]: Why this change?

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 5]
```diff
@@ -17,6 +17,11 @@ export class CourseItemsListingView {
       $section.find('.open-response-assessment-msg').show();
     }
 
+    const context = this.data.CONTEXT || {};
+
+    const esgEnabled = context.ENHANCED_STAFF_GRADER;
+    const esgUrl = context.ORA_GRADING_MICROFRONTEND_URL;
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:23]: Naming nit: this should probably be base/root URL. URL by itself suggests to me a complete path.

```suggestion
    const esgRootUrl = context.ORA_GRADING_MICROFRONTEND_URL;
```

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 6]
```diff
@@ -28,9 +33,21 @@ export class CourseItemsListingView {
         const hasAssessmentType = this.model.get(this.type ? this.type : 'staff_assessment');
         let link = null;
         if (itemViewEnabled && (!this.type || (this.type && hasAssessmentType))) {
+          const displayValue = this.title || formattedValue;
+          if (esgEnabled) {
+            const id = this.model.get('id');
+            link = $('<a>', {
+              text: displayValue,
+              title: displayValue,
+              href: `${esgUrl}/${id}`,
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:42]: We have control of both `esgUrl` and `id` so it is less of an issue, but best practice to use a URL join/create instead of string building here.

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 7]
```diff
@@ -28,9 +33,21 @@ export class CourseItemsListingView {
...
+              class: 'staff-esg-link',
+            });
+            this.$el.append(link);
+            return this;
+          }
           link = $('<a>', {
-            text: formattedValue,
-            title: this.title || formattedValue,
+            text: displayValue,
+            title: displayValue,
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:50]: ⚠️ This looks like a change in behavior. Is this intentional?

**Review by nsprenkle — APPROVED** [review]
Looks good but haven't tested locally. Added some review notes which you can take or ignore.

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 8]
```diff
@@ -126,6 +143,19 @@ export class CourseItemsListingView {
...
+        name: 'response',
+        label: gettext('Response'),
+        label_summary: gettext('Response'),
+        cell: StaffCell.extend({
+          title: gettext('View and grade responses'),
+        }),
+        num: true,
+        editable: false,
+      });
+    }
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:158]: Yeah, this bit doesn't look like it's tested. Shouldn't it be?

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js (hunk 7 above)
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:50]: Tested there isn't any thing change. Title doesn't get define except the additional column I add.

File: openassessment/staffgrader/tests/test_decorators.py (hunk 4 above)
Comment (leangseu-edx) [reply openassessment/staffgrader/tests/test_decorators.py:10]: pylint

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js (hunk 6 above)
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/lms/oa_course_items_listing.js]: this code doesn't use node. There isn't path join I can use.
